Tools/Scripts/libraries/webkitbugspy/webkitbugspy/github.py

```diff
@@ -66,14 +66,26 @@ def from_string(self, string):
         return None
 
     def credentials(self, required=True):
-        return credentials(
+        hostname = self.url.split('/')[2]
+        args = dict(
             url=self.api_url,
             required=required,
             name=self.url.split('/')[2].replace('.', '_').upper(),
-            prompt="GitHub's API\nPlease generate a 'Personal access token' via 'Developer settings' for your user",
+            prompt='''GitHub's API
+Please go to https://{host}/settings/tokens and generate a new 'Personal access token' via 'Developer settings'
+with 'repo' and 'workflow' access and appropriate 'Expiration' for your {host} user'''.format(host=hostname),
             key_name='token',
         )
 
+        username, token = webkitcorepy.credentials(**args)
+        if username and '@' in username:
+            sys.stderr.write("Provided username contains an '@' symbol. Please make sure to enter your GitHub username, not an email associated with the account\n")
+            webkitcorepy.delete_credentials(url=args['url'], name=args['name'])
```

Comment:
Instead of deleting from keychain after we detect '@' in username, can we simply skip adding to keychain?

Reply:
No, because then `webkitcorepy` would have to know that `@` was not allowed in usernames, and the trouble is that's not generally true. Notably, it's not true for bugzilla.